### DISCUSSION TITLE
Analyse a raw string rather than files/directories?

### DISCUSSION THREAD

**Discussion by lowlighter** [Nixinova/LinguistJS#9] [Questions] [answered]
Hi, thanks a lot for this tool 🙂 !

From what I understand, the exported `analyse` function only takes a list of paths as inputs, and from there it'll walk on each file in given directories:
https://github.com/Nixinova/Linguist/blob/9a15a06266b45a0075351c8884a519d42cc252b0/src/index.ts#L16-L18

Would it be possible to expose another entry point which directly accepts a raw string as argument and returns the result instead ?

There are some cases where the content is already in memory (like when fetching from APIs) so it would avoid saving it on disk and save some I/O

**Comment by Nixinova** [comment]
That sounds good, although it would require a lot of refactoring. I'll get to it later.

**Reply by Nixinova** [reply]
Tracking in #10

**Reply by lowlighter** [reply]
Cool! Thanks a lot 👍

**Comment by Nixinova** [comment, marked as answer]
@lowlighter does the addition in 205c4ef461bd8a16b056bbf21485cc3689be5829 cover your use case? (and does this look like the best way of doing this?) If so i'll release it as v2.3

```js
//example//
const linguist = require('linguist-js')
linguist('filename', {fileContent: 'whatever'})
linguist(['file1', 'file2'], {fileContent: ['foo','bar']})
```

**Reply by lowlighter** [reply]
Awesome!
Yes it should cover my use case 👍
